File: flutter_app/lib/ui/leaderboard_screen.dart
import 'package:flutter/material.dart';
import '../core/palette.dart';
import '../data/entities.dart';

class LeaderboardScreen extends StatelessWidget {
  final List<RunResult> topRuns;
  final VoidCallback onBackClicked;

  const LeaderboardScreen({
    required this.topRuns,
    required this.onBackClicked,
    super.key,
  });

  @override
  Widget build(BuildContext context) {
    return Container(
      color: Palette.voidBlack,
      padding: const EdgeInsets.all(20),
      child: Column(
        children: [
          const Text("HIGH SCORES", style: TextStyle(color: Palette.gold, fontSize: 24, fontFamily: 'monospace', fontWeight: FontWeight.w900)),
          const SizedBox(height: 16),
          Expanded(
            child: topRuns.isEmpty
                ? const Center(
                    child: Text("No runs recorded yet!\nStart playing to set high scores.",
                        textAlign: TextAlign.center, style: TextStyle(color: Palette.bone, fontSize: 14, fontFamily: 'monospace')),
                  )
                : ListView.builder(
                    itemCount: topRuns.length,
                    itemBuilder: (context, index) {
                      final run = topRuns[index];
                      return Container(
                        margin: const EdgeInsets.symmetric(vertical: 6),
                        padding: const EdgeInsets.all(12),
                        decoration: BoxDecoration(
                          color: Palette.shadow,
                          borderRadius: BorderRadius.circular(6),
                          border: Border.all(color: Palette.slate, width: 1),
                        ),
                        child: Row(
                          mainAxisAlignment: MainAxisAlignment.spaceBetween,
                          children: [
                            Text("#${index + 1}  ${run.distanceMeters}m",
                                style: TextStyle(
                                  color: index == 0 ? Palette.gold : Palette.paper,
                                  fontSize: 16,
                                  fontFamily: 'monospace',
                                  fontWeight: FontWeight.bold,
                                )),
                            Text("Taka${run.moneyCollected} (${run.zoneReached})",
                                style: const TextStyle(color: Palette.amber, fontSize: 14, fontFamily: 'monospace')),
                          ],
                        ),
                      );
                    },
                  ),
          ),
          const SizedBox(height: 16),
          _buildButton("BACK", Palette.slate, Palette.paper, onBackClicked),
        ],
      ),
    );
  }

  Widget _buildButton(String text, Color bg, Color textColor, VoidCallback onTap) {
    return GestureDetector(
      onTap: onTap,
      child: Container(
        width: double.infinity,
        height: 50,
        decoration: BoxDecoration(
          color: bg,
          borderRadius: BorderRadius.circular(6),
          border: Border.all(color: Palette.ink, width: 2),
        ),
        child: Center(
          child: Text(text, style: TextStyle(color: textColor, fontSize: 16, fontFamily: 'monospace', fontWeight: FontWeight.bold)),
        ),
      ),
    );
  }
}

File: app/src/main/java/com/example/ui/LeaderboardScreen.kt
package com.example.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.itemsIndexed
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.core.Palette
import com.example.data.RunResultEntity

@Composable
fun LeaderboardScreen(
    topRuns: List<RunResultEntity>,
    onBackClicked: () -> Unit
) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Palette.voidBlack)
            .padding(20.dp),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Text(
            text = "🏆 HIGH SCORES",
            color = Palette.gold,
            fontSize = 24.sp,
            fontWeight = FontWeight.Black,
            fontFamily = FontFamily.Monospace,
            modifier = Modifier.padding(vertical = 16.dp)
        )

        if (topRuns.isEmpty()) {
            Box(
                modifier = Modifier
                    .weight(1f)
                    .fillMaxWidth(),
                contentAlignment = Alignment.Center
            ) {
                Text(
                    text = "No runs recorded yet!\nStart playing to set high scores.",
                    color = Palette.bone,
                    fontSize = 14.sp,
                    fontFamily = FontFamily.Monospace
                )
            }
        } else {
            LazyColumn(
                modifier = Modifier.weight(1f),
                verticalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                itemsIndexed(topRuns) { index, run ->
                    Box(
                        modifier = Modifier
                            .fillMaxWidth()
                            .background(Palette.shadow, RoundedCornerShape(6.dp))
                            .border(1.dp, Palette.slate, RoundedCornerShape(6.dp))
                            .padding(12.dp)
                    ) {
                        Row(
                            modifier = Modifier.fillMaxWidth(),
                            horizontalArrangement = Arrangement.SpaceBetween,
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            Text(
                                text = "#${index + 1}  ${run.distanceMeters}m",
                                color = if (index == 0) Palette.gold else Palette.paper,
                                fontSize = 16.sp,
                                fontWeight = FontWeight.Bold,
                                fontFamily = FontFamily.Monospace
                            )
                            Text(
                                text = "৳${run.moneyCollected} (${run.zoneReached})",
                                color = Palette.amber,
                                fontSize = 14.sp,
                                fontFamily = FontFamily.Monospace
                            )
                        }
                    }
                }
            }
        }

        Spacer(modifier = Modifier.height(16.dp))

        RetroArcadeButton(
            text = "⬅  B A C K",
            bgColor = Palette.slate,
            textColor = Palette.paper,
            testTag = "back_button",
            onClick = onBackClicked
        )
    }
}
